core/iwasm/common/wasm_native.c

```diff
@@ -394,6 +407,155 @@ wasm_native_unregister_natives(const char *module_name,
     return false;
 }
 
+static uint32
+context_key_to_idx(void *key)
```

Comment:
Suggest to make this feature configurable with cmake variable and compiler macro, and it is enabled only when libc wasi is enabled? Or just using `cmake -DWAMR_BUILID_LIBC_WASI/UVWASI=1` to control it? Since the feature may introduce more memory usage and binary size, some host embedder may expect the footprint as small as possible.

Reply:
i can make it a build-time option if desirable.

Reply:
I'd like to control wasi-nn context with this API for both libc wasi and libc builtin.
I think that the feature can also be helpful for a generic native library.